src/main/java/com/google/sps/servlets/DeleteMarker.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that process information about user's markers. */
+@SuppressWarnings("serial")
+@WebServlet("/delete_marker")
+public class DeleteMarker extends HttpServlet {
+  /** Processes POST request by deleting marker by received id. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get Id from the request.
+    try {
+      String contentId = request.getParameter("contentId");
+      Long id = Long.parseLong(contentId);
+
+      Key markerKey = KeyFactory.createKey("Marker", id);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.delete(markerKey);
+    } catch (Exception e) {
+      System.out.println("Can't find marker by id");
```

Comment:
Where would this message be printed? I thinks that maybe is better to print errors on the response page, so that the user will also see what the problem was.

Reply:
+1. It would be better to return a status to the client whether the deletion was successful or not, which you can then display on the client.

Also, instead of using `System.out.println` for things on the server, some form of logging should be used if you want to notify someone on the server. In a production environment nobody will be looking at the console output while the web site is running. If errors are logged to a file or a loggin system, automated processes can be set up for monitoring and alters. Google App Engine provides a logging API. See https://cloud.google.com/appengine/docs/standard/java/logs and https://cloud.google.com/logging/docs/setup/java

Data displayed to a client should not contain technical details, so this message would be good to send to a client. When logging on the server, the message should also contain the exception details to allow for troubleshooting. An administrator reading this error in the logs will have no idea why the error happened or how to fix it.

Rather than making this PR even larger, perhaps create an issue to do this and add a `TODO` comment to the code to do is as a follow-up PR.

Reply:
Issue created